File: app/src/main/java/com/vktiwari/pagingretrofitrxjava/model/StarWarResponse.java
package com.vktiwari.pagingretrofitrxjava.model;

import android.text.TextUtils;

import java.io.Serializable;
import java.util.List;

public class StarWarResponse<T> implements Serializable {
    public int count;
    public String next;
    public String previous;
    public List<T> results;

    public boolean hasMore() {
        return !TextUtils.isEmpty(next);
    }
}

File: app/src/kotlin/kotlin/com/vktiwari/pagingretrofitrxjava/model/StarWarResponse.kt
package com.vktiwari.pagingretrofitrxjava.model;

import android.text.TextUtils
import java.io.Serializable

public class StarWarResponse<T> : Serializable {
    var count: Int = 0
    var next: String? = null
    var previous: String? = null
    var results: kotlin.collections.List<T>? = null

    fun hasMore(): Boolean {
        return !TextUtils.isEmpty(next)
    }
}
